src/workerd/jsg/fast-api.h

```diff
@@ -48,7 +55,8 @@ concept FastApiPrimitive = kj::isSameType<T, void>() || kj::isSameType<T, bool>(
 
 // Helper to determine if a type can be used as a parameter in V8 Fast API
 template <typename T>
-concept FastApiParam = !isFunctionCallbackInfo<kj::RemoveConst<kj::Decay<T>>>;
+concept FastApiParam = !isFunctionCallbackInfo<kj::RemoveConst<kj::Decay<T>>> &&
+    !isKjPromise<kj::RemoveConst<kj::Decay<T>>>;
```

Comment:
`RemoveConst<Decay<T>>` is redundant, `Decay` already removes const.

Reply:
Good to know. Will do a follow up. Thanks!